dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -15,266 +15,181 @@
  */
 package io.dockstore.webservice.resources;
 
-import com.github.zafarkhaja.semver.Version;
-import com.google.common.collect.Lists;
 import io.dockstore.webservice.CustomWebApplicationException;
 import io.dockstore.webservice.core.Entry;
 import io.dockstore.webservice.core.Organization;
 import io.dockstore.webservice.core.Token;
 import io.dockstore.webservice.core.User;
 import java.util.List;
 import java.util.Optional;
-import javax.ws.rs.container.ContainerRequestContext;
 import org.apache.http.HttpStatus;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Endpoints that use authentication by Dockstore user
+ * AuthenticatedResourceInterface is a mixin that provides methods used to implement user-level access control in
+ * Resource handlers.  By using these methods consistently, we centralize access-checking logic and avoid its
+ * repetition, allowing us to easily modify our access policies, if necessary.
+ *
+ * Each of the "check" methods returns on success, and throws an appropriate CustomWebApplicationException on failure.
+ * The most commonly-used "check" methods check if a user is allowed to perform a type of action on a specified entry:
+ *
+ * <ul>
+ * <li>checkCanRead: can the user read the specified entry?
+ * <li>checkCanWrite: can the user write (modify) the specified entry?
+ * <li>checkCanShare: can the user share (publish) the specified entry?
+ * </ul>
+ *
+ * There are a number of "check" utility methods that perform other useful checks:
+ *
+ * <ul>
+ * <li>checkIsOwner: is the user one of the users that controls the specified entry?
```

Comment:
Wonder if this is actually needed. It seems like this is only used by `@Path("/{containerId}/users")` and doesn't seem to have a workflow equivalent (or maybe it wasn't affected by this PR)

Reply:
I changed the workflow equivalent so it also won't return a list of users to non-users (or SAM authorized useds).  Is that the desired behavior?